src/main/java/com/amazonaws/kinesisvideo/parser/examples/lambda/H264FrameProcessor.java

```diff
@@ -132,38 +132,44 @@ public static H264FrameProcessor create(final AWSCredentialsProvider credentials
     public void process(final Frame frame, final MkvTrackMetadata trackMetadata,
                         final Optional<FragmentMetadata> fragmentMetadata) throws FrameProcessException {
         if (rekognizedOutputs != null) {
-
-            checkState(fragmentMetadata.isPresent(), "FragmentMetadata should be present !");
-
-            // Decode H264 frame
-            final BufferedImage decodedFrame = h264Decoder.decodeH264Frame(frame, trackMetadata);
-            log.debug("Decoded frame : {} with timecode : {} and fragment metadata : {}",
-                    frameNo, frame.getTimeCode(), fragmentMetadata.get());
-
-            // Get Rekognition results for this fragment number
-            final Optional<RekognizedOutput> rekognizedOutput = findRekognizedOutputForFrame(frame, fragmentMetadata);
-
-            // Render frame with bounding box
-            final BufferedImage compositeFrame = renderFrame(decodedFrame, rekognizedOutput);
-
-            // Encode to H264 frame
-            final EncodedFrame encodedH264Frame = encodeH264Frame(compositeFrame);
-            encodedH264Frame.setTimeCode(fragmentMetadata.get().getProducerSideTimestampMillis() + frame.getTimeCode());
-            log.debug("Encoded frame : {} with timecode : {}", frameNo, encodedH264Frame.getTimeCode());
-
-            // Call PutFrame for processed encodedFrame.
-            putFrame(encodedH264Frame);
-            frameNo++;
-
+            // Process only for video frames
+            if (frame.getTrackNumber() == VIDEO_TRACK_NO) {
+                checkState(trackMetadata.getPixelWidth().isPresent() && trackMetadata.getPixelHeight().isPresent(),
+                        "Missing video resolution in track metadata !");
+                checkState(fragmentMetadata.isPresent(), "FragmentMetadata should be present !");
+
+                // Decode H264 frame
+                final BufferedImage decodedFrame = h264Decoder.decodeH264Frame(frame, trackMetadata);
+                log.debug("Decoded frame : {} with timecode : {} and fragment metadata : {}",
```

Comment:
/nit frame Number :{}

Reply:
should we consider excluding the log4j dependency from   
         <dependency>
            <artifactId>aws-java-sdk-kinesis</artifactId>
            <groupId>com.amazonaws</groupId>
        </dependency>

 through https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/blob/master/pom.xml

Reply:
We will need the log4j dependency in Java SDK